dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -218,4 +230,257 @@ public Response deleteEnvironmentPushHistory(@Context HttpServletRequest request
+                restResponse = Response.ok(new ResponseEntityView(
+                        CollectionsUtils.map("bundlesDeleted", deleteBundlesByIdentifierForm.getIdentifiers()))
+                        ).build();
+                asyncResponse.resume(restResponse);
+            } catch (DotDataException e) {
+
+                Logger.error(this.getClass(),
+                        "Exception on deleteBundlesByIdentifiers, couldn't delete the identifiers: "
+                                + deleteBundlesByIdentifierForm.getIdentifiers() +
+                                ", exception message: " + e.getMessage(), e);
+                restResponse = ResponseUtil.mapExceptionResponse(e);
+                asyncResponse.resume(restResponse);
+            }
+        });
+    } // deleteBundlesByIdentifiers.
+
+    // one transaction for each bundle
+    private void deleteBundleByIdentifier(final DeleteBundlesByIdentifierForm deleteBundlesByIdentifierForm, final InitDataObject initData) throws DotDataException {
+
+        for (final String bundleId : deleteBundlesByIdentifierForm.getIdentifiers()) {
+
+            this.bundleAPI.deleteBundleAndDependencies(bundleId, initData.getUser());
+        }
+    }
+
+    /**
+     * Deletes bundles older than a date. (unsent are not going to be deleted)
+     * Note: The process could be heavy, so it is handle by async response
+     * @param request
+     * @param response
+     * @param asyncResponse response is async
+     * @param olderThan
+     */
+    @DELETE
+    @Path("/olderthan/{olderThan}")
+    @Produces("application/json")
+    public void deleteBundlesOlderThan(@Context   final HttpServletRequest request,
+                                       @Context   final HttpServletResponse response,
+                                       @Suspended final AsyncResponse asyncResponse,
+                                       @PathParam("olderThan") final DateParam olderThan) {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        Logger.info(this, "Deleting the bundles older than: " + olderThan
+                + " by the user: " + initData.getUser().getUserId());
+
+        final DotSubmitter dotSubmitter = DotConcurrentFactory
+                .getInstance().getSubmitter(BUNDLE_THREAD_POOL_SUBMITTER_NAME);
+        dotSubmitter.execute(() -> {
+
+            Response restResponse = null;
+
+            try {
+
+                restResponse = Response.ok(new ResponseEntityView(CollectionsUtils.map("bundlesDeleted",
+                        this.bundleAPI.deleteBundleAndDependenciesOlderThan(olderThan, initData.getUser())))).build();
+                asyncResponse.resume(restResponse);
+            } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)